apps/condo/domains/news/components/RecipientCounter.tsx

```diff
@@ -155,72 +136,62 @@ export const RecipientCounter: React.FC<RecipientCounterProps> = ({ newsItemScop
-            deletedAt: null,
-        },
-    })
-
-    const { objs: residents, loading: loadingResidents } = Resident.useAllObjects({
-        where: queryFindResidentsByOrganizationAndScopes(organization.id, newsItemScopes),
-    })
-
-    const processedNewsItemScope = newsItemScopes.reduce<TNewsItemScopeNoInstance[]>((acc, scope) => {
-        const partialScope: TNewsItemScopeNoInstance = {}
-        if (get(scope, 'property')) partialScope.property = scope.property
-        if (get(scope, 'unitType')) partialScope.unitType = scope.unitType
-        if (get(scope, 'unitName')) partialScope.unitName = scope.unitName
-        if (!isEmpty(partialScope)) acc.push(partialScope)
-        return [...acc]
-    }, [])
+            return [
+                ...acc,
+                {
+                    property: property ? { id: property.id } : null,
+                    unitType: get(scope, 'unitType'),
+                    unitName: get(scope, 'unitName'),
+                },
+            ]
+        }, [])
+    }, [newsItemScopes])
 
-    const unitsAffected = useMemo<IAffectedUnit[]>((): IAffectedUnit[] => {
-        if (loadingProperties || !properties || loadingResidents || !residents) {
-            return []
-        }
+    const [getCountersMutation] = useMutation(GET_NEWS_ITEMS_RECIPIENTS_COUNTERS_MUTATION)
+    useEffect(() => {
+        setIsCountersLoading(true)
+        const sender = getClientSideSenderInfo()
+        const meta = { dv: 1, sender }
 
-        if (processedNewsItemScope.length === 0) {
-            return properties.reduce<IAffectedUnit[]>((result, property) => {
-                return [
-                    ...result,
-                    ...getUnitsFromProperty(property).map<IAffectedUnit>((unit) => ({
-                        propertyId: property.id,
-                        type: NewsItemScopeUnitTypeType.Flat,
-                        name: unit,
-                    })),
-                ]
-            }, [])
-        } else {
-            return processedNewsItemScope.reduce<IAffectedUnit[]>((result, scope) => {
-                if (scope.unitType && scope.unitName) {
-                    return [...result, { propertyId: scope.property.id, type: scope.unitType, name: scope.unitName }]
-                } else {
-                    const unitNames = getUnitsFromProperty(properties.find((property) => property.id === scope.property.id))
-                    return [
-                        ...result,
-                        ...unitNames.map<IAffectedUnit>((unitName) => ({
-                            propertyId: scope.property.id,
-                            type: NewsItemScopeUnitTypeType.Flat,
-                            name: unitName,
-                        }))]
-                }
-            }, [])
-        }
-    }, [processedNewsItemScope, properties, residents, loadingProperties, loadingResidents])
+        runMutation({
```

Comment:
What is the main reason of `runMutation` usage here? Did your backend service send some special user-friendly errors, that you need to display? For me best option here will be `useLazyQuery` hook, that also provide `loading` option

Reply:
You right. Refactored using `useLazyQuery`.
Thanks for the good comment!